ext-src/php_swoole_cxx.h

```diff
@@ -107,18 +107,7 @@ SW_API bool php_swoole_timer_clear(swoole::TimerNode *tnode);
 SW_API bool php_swoole_timer_clear_all();
 
 static inline bool php_swoole_is_fatal_error() {
-    if (PG(last_error_message)) {
-        switch (PG(last_error_type)) {
-        case E_ERROR:
-        case E_CORE_ERROR:
-        case E_USER_ERROR:
-        case E_COMPILE_ERROR:
-            return true;
-        default:
-            break;
-        }
-    }
-    return false;
+    return PG(last_error_type) & E_FATAL_ERRORS;
```

Comment:
```suggestion
    return PG(last_error_message) && (PG(last_error_type) & E_FATAL_ERRORS);
```


see: https://github.com/php/php-src/blob/fabfe1e3342f691065b6b6405a8026ba09cc592e/sapi/cli/php_cli_server.c#L1177-L1178